### PR TITLE
ENH: ICE calendar, more calendar tests, cleaned up CME calendar

### PR DESCRIPTION
**Pull request by jbredeche**
Added tests for CME calendar.

Added ICE calendar (and tests).

### CODE REVIEW

File: zipline/utils/calendars/exchange_calendar_ice.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                USLaborDay,
+                USThanksgivingDay
+            ]))
+        ]
+
+    @property
+    def adhoc_holidays(self):
+        # ICE was only closed on the first day of the Hurricane Sandy closings
+        # (was not closed on 2012-10-30)
+        return [Timestamp('2012-10-29', tz='UTC')]
```
Comment (yankees714) [inline zipline/utils/calendars/exchange_calendar_ice.py]: Just want to verify, CME includes `USNationalDaysofMourning` here, but ICE doesn't?

File: zipline/utils/calendars/trading_calendar.py [hunk 2]
```diff
@@ -176,7 +256,7 @@ def next_open(self, dt):
             The UTC timestamp of the next open.
         """
         idx = next_divider_idx(self.market_opens_nanos, dt.value)
-        return self.schedule.market_open[idx].tz_localize('UTC')
+        return pd.Timestamp(self.market_opens_nanos[idx], tz='UTC')
```
Comment (yankees714) [inline zipline/utils/calendars/trading_calendar.py]: What's this change for?

File: zipline/utils/calendars/trading_calendar.py [hunk 3]
```diff
@@ -645,15 +725,15 @@ def _special_dates(self, calendars, ad_hoc_dates, start_date, end_date):
 
     def _special_opens(self, start, end):
         return self._special_dates(
-            self.special_opens_calendars,
+            self.special_opens,
```
Comment (yankees714) [inline zipline/utils/calendars/trading_calendar.py]: Seems a little weird that we're using `self.special_opens` within `self._special_opens`, maybe there's a better name for one of these?

File: tests/calendars/test_ice_calendar.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            next_day = pd.Timestamp(date_pair[1], tz='UTC')
+            the_open = self.calendar.schedule.loc[next_day].market_open
+
+            localized_open = the_open.tz_localize("UTC").tz_convert(
+                self.calendar.tz
+            )
+
+            self.assertEqual(
+                date_pair[0],
+                (localized_open.year, localized_open.month, localized_open.day)
```
Comment (yankees714) [inline tests/calendars/test_ice_calendar.py]: Probably worth checking the time here too

File: zipline/utils/calendars/trading_calendar.py (hunk 2 above)
Comment (jbredeche) [inline zipline/utils/calendars/trading_calendar.py]: sorry, I should have commented, or put it in a separate commit - this is a performance improvement.  The new line runs ~8x faster than the old line.

File: zipline/utils/calendars/exchange_calendar_ice.py (hunk 1 above)
Comment (jbredeche) [inline zipline/utils/calendars/exchange_calendar_ice.py]: good catch.  I can't find an explicit mention online of ICE being closed for Reagan and Ford's deaths, but I found enough hints of "other futures markets will do the same" that it's a safe assumption that ICE was closed.

File: zipline/utils/calendars/trading_calendar.py (hunk 3 above)
Comment (jbredeche) [inline zipline/utils/calendars/trading_calendar.py]: I renamed `_special_opens` to `_calculate_special_opens`, etc.  Hope that helps.

File: tests/calendars/test_ice_calendar.py (hunk 4 above)
Comment (jbredeche) [inline tests/calendars/test_ice_calendar.py]: Agreed.  Added hour/minute check, and then moved this test to the base test suite so it's hit in all the calendar tests.
